offer-manager-domain/src/main/groovy/life/qbic/business/offers/OfferContent.groovy

```diff
@@ -316,4 +316,15 @@ class OfferContent {
         totalVat = builder.totalVat
     }
 
+    List<OfferItem> getDataGenerationItems() {
+        return Collections.unmodifiableList(dataGenerationItems)
+    }
```

Comment:
You do not need to return an unmodifiable list only make sure the list in this object cannot be modified.
```suggestion
    List<OfferItem> getDataGenerationItems() {
        return dataGenerationItems.collect()
    }
```

Reply:
This way you return a new instance of a list with the same content and changes to the returned list do not affect the list of your offer content

Reply:
what speaks against my suggestion?

Reply:
in general its not wrong, right?

Reply:
I think it is wrong to return an unmodifiable list. It restricts the consumer of this class in what can be done with it and forces him to copy the whole list again. In the end you would end up with up to 3 lists with the same information